### PR TITLE
[WIP]Update News Lease

### PR DESCRIPTION
**Pull request by iamPhong**
- [x] update news lease

- [ ] rspec test

### CODE REVIEW

**Comment by iamPhong** [comment]
Ready

File: app/assets/javascripts/guest/news_leases.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      document.getElementById('lng').value = place.geometry.location.lng();
+      bounds.extend(place.geometry.location);
+      marker.setPosition(place.geometry.location);
+    }
+    map.fitBounds(bounds);
+    map.setZoom(15);
+  });
+}
+
+$(document).ready(function() {
```
Comment (TruongQuocDat) [inline app/assets/javascripts/guest/news_leases.js]: Why don't write this code to new function JS with name preview image. You should remember rule DRY
This code was repeat so more. It not good.

File: app/assets/javascripts/guest/news_leases.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  var main_img = $('#main_img');
+  var sub_img = $('#sub_img');
+
+  $('.upload-main-img').change(function(event) {
+    var input = $(event.currentTarget);
```
Comment (TruongQuocDat) [inline app/assets/javascripts/guest/news_leases.js]: Why don't write this code to new function JS with name preview image. You should remember rule DRY
This code was repeat so more. It not good.

File: app/assets/javascripts/custom.js [hunk 3]
```diff
@@ -20,14 +20,52 @@ $(document).ready(function() {
...
-     var input = $(event.currentTarget);
-     var file = input[0].files[0];
-     var reader = new FileReader();
-     reader.onload = function(e){
-         image_base64 = e.target.result;
-         preview.attr("src", image_base64);
-     };
-     reader.readAsDataURL(file);
+  $(".cfi-common-upload").change(function(event) {
+    var input = $(event.currentTarget);
```
Comment (TruongQuocDat) [inline app/assets/javascripts/custom.js]: Why don't write this code to new function JS with name preview image. You should remember rule DRY
This code was repeat so more. It not good.

File: app/assets/javascripts/guest/news_leases.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    var reader = new FileReader();
+    reader.onload = function(e) {
+      image_base64 = e.target.result;
+      main_img.attr('src', image_base64);
+    };
+    reader.readAsDataURL(file);
+  });
+
+  $('.upload-sub-img').change(function(event) {
+    var input = $(event.currentTarget);
```
Comment (TruongQuocDat) [inline app/assets/javascripts/guest/news_leases.js]: Why don't write this code to new function JS with name preview image. You should remember rule DRY
This code was repeat so more. It not good.

File: app/controllers/guest/news_leases_controller.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
+class Guest::NewsLeasesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_lease, except: %i(index new create)
```
Comment (TruongQuocDat) [inline app/controllers/guest/news_leases_controller.rb]: Where is your index method?

**Comment by huongnguyenmta** [comment]
conflicted

**Comment by iamPhong** [comment]
Ready

File: app/controllers/guest/news_leases_controller.rb [hunk 6]
```diff
@@ -19,9 +20,30 @@ def create
     end
   end
 
+  def edit
+    @category = Category.all
```
Comment (huongnguyenmta) [inline app/controllers/guest/news_leases_controller.rb]: em get thông tin này để làm gì đây em??

File: app/controllers/guest/news_leases_controller.rb [hunk 7]
```diff
@@ -19,9 +20,30 @@ def create
...
   def lease_params
     params.require(:news_lease).permit(:category_id, :user_id, :price_type, :price_value, :deposit_price, :deposit_paper,
     :product_name, :description, :sell, :key_search, :lease_name, :lease_phone_number, place_attributes: [:lat, :lng, :address],
     image_attributes: [:main_photo, :sub_photo])
   end
+
+  def set_lease
+    @lease = NewsLease.find_by id: params[:id]
+    return if @lease
+    flash[:danger] = "Không tìm thấy"
```
Comment (huongnguyenmta) [inline app/controllers/guest/news_leases_controller.rb]: I18n

File: app/helpers/guest/news_leases_helper.rb [hunk 8]
```diff
@@ -1,2 +1,11 @@
-module Guest::NewsLeasesHelper
-end
+module Guest::NewsLeasesHelper
+  def format_price(price_type)
+    if price_type == 0
+      return t "day"
```
Comment (huongnguyenmta) [inline app/helpers/guest/news_leases_helper.rb]: bỏ từ `return` đi em

File: app/views/guest/news_leases/_form.html.erb [hunk 9]
```diff
@@ -114,70 +131,75 @@
         </div>
         <div></div>
         <div id="lease-map" class="" style="height: 300px; position: relative; overflow: hidden;"></div>
-        <%= o.hidden_field :lat, value:"21.027451", id:"lat" %>
-        <%= o.hidden_field :lng, value:"105.834169", id:"lng" %>
+        <% if @lease.place.present? %>
+          <%= o.hidden_field :lat, value:@lease.place.lat, id:"lat" %>
+          <%= o.hidden_field :lng, value:@lease.place.lng, id:"lng" %>
+        <% else %>
+          <%= o.hidden_field :lat, value:"21.027451", id:"lat" %>
```
Comment (huongnguyenmta) [inline app/views/guest/news_leases/_form.html.erb]: value cho vào settings đi em

File: spec/controllers/guest/news_leases_controller_spec.rb [hunk 10]
```diff
@@ -43,4 +43,21 @@
...
     end
   end
+
+  describe "UPDATE" do
+    context "with valid attributes" do
+      @lease = FactoryBot.create(:news_lease)
+      let(:attr) do
+        { :description => 'new', :key_search => 'new key' }
+      end
+      byebug
```
Comment (huongnguyenmta) [inline spec/controllers/guest/news_leases_controller_spec.rb]: byebug nài -_-

File: app/controllers/guest/news_leases_controller.rb (hunk 6 above)
Comment (iamPhong) [reply app/controllers/guest/news_leases_controller.rb]: Trong view edit em có render ra các category ạ

**Comment by iamPhong** [comment]
Ready

**Pull request merged**